### PR TITLE
[DOCS]  Disabling guest api checkout

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12          |
| License         | MIT                                                          |

### CODE REVIEW

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 1]
```diff
@@ -19,6 +19,44 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
+
+.. code-block:: yaml
+
+    # config/packages/security.yaml
+    security:
+        access_control:
+    -       - { path: "%sylius.security.new_api_shop_regex%/.*", role: IS_AUTHENTICATED_ANONYMOUSLY }
+    +       - { path: "%sylius.security.new_api_shop_regex%/orders", method: POST, GET , role: IS_AUTHENTICATED_ANONYMOUSLY }
+    +       - { path: "%sylius.security.new_api_shop_regex%/orders/.*/items", method: POST , role: IS_AUTHENTICATED_ANONYMOUSLY }
+    +       - { path: "%sylius.security.new_api_shop_regex%/.*", role: ROLE_USER }
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]:
```suggestion
    # config/packages/security.yaml
    security:
        access_control:
            - { path: "%sylius.security.new_api_shop_regex%/.*", role: IS_AUTHENTICATED_ANONYMOUSLY }
            - { path: "%sylius.security.new_api_shop_regex%/orders", method: POST, GET , role: IS_AUTHENTICATED_ANONYMOUSLY }
            - { path: "%sylius.security.new_api_shop_regex%/orders/.*/items", method: POST , role: IS_AUTHENTICATED_ANONYMOUSLY }
            - { path: "%sylius.security.new_api_shop_regex%/.*", role: ROLE_USER }
```

Comment (CoderMaggie) [reply docs/cookbook/shop/disabling-guest-checkout.rst]: something went crazy in here 👯

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 2]
```diff
@@ -19,6 +19,44 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
 will be redirected to the checkout addressing step.
 
+How to disable guest checkout in new API?
+=========================================
+
+Our new API let's any anonymous user to proceed through the checkout out of the box (of course if API is enabled).
+In order to disable checking out for anonymous user you have to change the same file but use different routes.
+Here we will let user to only:
+
+    * Get new cart
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]:
```suggestion
    * Pick up a new cart
```

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 3]
```diff
@@ -19,6 +19,44 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
+    * Get the order by token value
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]: 🤔 I do not get that?

```suggestion
    * Have access to it by its token value
```

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 4]
```diff
@@ -19,6 +19,44 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
+    * Let the user to add an item to the cart
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]:
```suggestion
    * Let the user add items to the cart
```

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 5]
```diff
@@ -19,6 +19,44 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
+    security:
+        access_control:
+    -       - { path: "%sylius.security.new_api_shop_regex%/.*", role: IS_AUTHENTICATED_ANONYMOUSLY }
+    +       - { path: "%sylius.security.new_api_shop_regex%/orders", method: POST, GET , role: IS_AUTHENTICATED_ANONYMOUSLY }
+    +       - { path: "%sylius.security.new_api_shop_regex%/orders/.*/items", method: POST , role: IS_AUTHENTICATED_ANONYMOUSLY }
+    +       - { path: "%sylius.security.new_api_shop_regex%/.*", role: ROLE_USER }
+
+.. warning::
+
+    The "main" path (in this example "%sylius.security.new_api_shop_regex%/.*" ) should be at the very end of the configuration
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]:
```suggestion
    The "main" path (in this example ``"%sylius.security.new_api_shop_regex%/.*"`` ) should be at the very end of the configuration
```

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 6]
```diff
@@ -19,6 +19,44 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
+    with the same route, otherwise this would not work.
+
+Now when anonymous user will try to use other routes he will be informed that he is not authenticated:
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]:
```suggestion
Now when an anonymous user will try to use other checkout routes he will be informed that he is not authenticated:
```

**Review by CoderMaggie — CHANGES_REQUESTED** [review]

File: docs/cookbook/shop/disabling-guest-checkout.rst (hunk 1 above)
Comment (arti0090) [reply docs/cookbook/shop/disabling-guest-checkout.rst]: Not really crazy - i just wanted to show like a git diff that first line to be removed and other added. But looks like it is not possible in this docs :/
I will still fix this up with suggestions.

File: docs/cookbook/shop/disabling-guest-checkout.rst [hunk 7]
```diff
@@ -19,6 +19,43 @@ That will do the trick. Now, when a guest user tries to click the checkout butto
...
+           - { path: "%sylius.security.new_api_shop_regex%/orders", method: POST, GET , role: IS_AUTHENTICATED_ANONYMOUSLY }
+           - { path: "%sylius.security.new_api_shop_regex%/orders/.*/items", method: POST , role: IS_AUTHENTICATED_ANONYMOUSLY }
+           - { path: "%sylius.security.new_api_shop_regex%/.*", role: ROLE_USER }
+
+.. warning::
+
+    The "main" path (in this example ``"%sylius.security.new_api_shop_regex%/.*"`` ) should be at the very end of the configuration
+    with the same route, otherwise this would not work.
+
+Now when an anonymous user will try to use other checkout routes he will be informed that he is not authenticated:
```
Comment (CoderMaggie) [inline docs/cookbook/shop/disabling-guest-checkout.rst]:
```suggestion
Now when an anonymous user will try to use other checkout routes they will be informed that they are not authenticated:
```

**Review by CoderMaggie — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Pull request merged**